File: demo/src/main/resources/static/js/app/WeatherService.js
'use strict'
angular.module('demo.services', []).factory('WeatherService', ["$http", "CONSTANTS", function($http, CONSTANTS) {
    var service = {};
    service.fetchWeatherDetails = function(cityName) {
        var url = CONSTANTS.getWeatherDetailsUrl  + cityName ;
        return $http.get(url);
    }
    service.generatePdf = function() {
        return $http.get(CONSTANTS.generatePdfUrl);
    }

    service.fetchWeatherDetailsByDate= function(startDate, EndDate) {
    var url = CONSTANTS.getFilteredByDateUrl  + startDate +"/"+ EndDate  ;
        return $http.get(url);

     }
    return service;
}]);

File: demo/src/main/java/com/example/demo/service/WeatherService.java
package com.example.demo.service;

import com.example.demo.entity.Weather;

import java.util.Date;
import java.util.List;

public interface WeatherService {


    Weather getWeatherDetails(String cityName);
    List<Weather> dowloadWeatherPDF();
    List <Weather> getWeatherDetailsByDate(Date startDate, Date EndDate) ;
}


